server/schedule/operator_controller.go

```diff
@@ -83,6 +92,61 @@ func (oc *OperatorController) Dispatch(region *core.RegionInfo) {
 	}
 }
 
+func (oc *OperatorController) getNextPushOperatorTime(step OperatorStep, now time.Time) int64 {
+	nextTime := slowNotifyInterval
+	switch step.(type) {
+	case TransferLeader, PromoteLearner:
+		nextTime = fastNotifyInterval
+	}
+	return now.Add(nextTime).UnixNano()
+}
+
+func (oc *OperatorController) pollNeedDispatchRegion() (r *core.RegionInfo, next bool) {
+	oc.Lock()
+	defer oc.Unlock()
+	if oc.opNotifierQueue.Len() == 0 {
+		return nil, false
+	}
+	item := oc.opNotifierQueue[0]
+	regionID := item.op.regionID
+	op, ok := oc.operators[regionID]
+	if !ok || op == nil {
+		heap.Pop(&oc.opNotifierQueue)
```

Comment:
I think you can first pop it, then push it back to the heap if it is still pending.